test/test_basic.py

```diff
@@ -167,6 +167,13 @@ def test_id_safety(request):
     s1.id = safe_surf_id
     assert s1.id == safe_surf_id
 
+    s2 = model.surfaces_by_id[2]
+    s2.id = None
+    assert s2.id == safe_surf_id + 1
+
+    s2.id = 2
+    assert s2.id == 2
+
```

Comment:
How would we expect the ID sequence to work if a set is deleted?

Scenario:

1. Delete volume with ID 3 in a model with volumes 1, 2, and 3.
2. Create a new, fresh volume set

Does the new volume set have ID 3 or 4?

Reply:
I think it gets `3` because we'll remove `3` from the ID space when it gets deleted

Reply:
Actually - no!  We don't have destructors that do clean up of ID space.  So ID `3` only gets released if/when we renumber volume with ID 3..  If we delete it, the number stays reserved.

Reply:
Do you think I should add a destructor for cleanup?

Reply:
Sorry, my use of the term "delete" could've been more specific here and saved us some discussion I think. `del` vs `DAGSet.delete`